onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -82,6 +82,11 @@ ProviderInfo_ROCM& GetProviderInfo_ROCM();
 class FuseAdd : public OpKernel {
  public:
   explicit FuseAdd(const OpKernelInfo& info) : OpKernel(info) {
+    // logic for testing that a session options config value can be read here
+    auto test_throw_in_ctor = info.GetConfigOptions().GetConfigEntry("ThrowInKernelCtor");
+    if (test_throw_in_ctor.has_value() && test_throw_in_ctor.value() == "1") {
```

Comment:
nit: calling `optional::value` after `optional::has_value` will check for a value twice. `optional::operator*` is unchecked.
but here, we can probably just do a direct equality check since an empty optional will not be equal to a value.
```suggestion
    if (test_throw_in_ctor == "1") {
```

Reply:
Very cool - didn't know that was an option.